### PR TITLE
COMPAT: Ensure rolling indexers return intp during take operations

### LINKED ISSUE #35148: CI: MacPython failing test_consistency_with_window
https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=38672&view=logs&j=c0130b29-789d-5a3c-6978-10796a508a7f&t=e120bc6c-1f5e-5a41-8f0a-1d992cd2fbfb&l=1286

```python
    def test_consistency_with_window():
    
        # consistent return values with window
        df = test_frame
                index_array = self.index_array
            indexer = self.rolling_indexer(
                index_array=index_array, window_size=self.window_size,
            )
            start, end = indexer.get_window_bounds(
                len(indicies), min_periods, center, closed
            )
            start = start.astype(np.int64)
            end = end.astype(np.int64)
            # Cannot use groupby_indicies as they might not be monotonic with the object
            # we're rolling over
            window_indicies = np.arange(
                window_indicies_start, window_indicies_start + len(indicies),
            )
            window_indicies_start += len(indicies)
            # Extend as we'll be slicing window like [start, end)
            window_indicies = np.append(
                window_indicies, [window_indicies[-1] + 1]
            ).astype(np.int64)
>           start_arrays.append(window_indicies.take(start))
E           TypeError: Cannot cast array data from dtype('int64') to dtype('int32') according to the rule 'safe'

test_venv\lib\site-packages\pandas\core\window\indexers.py:318: TypeError
```

cc @mroeschke if you know where to look / who to ping. I haven't looked yet.

### LINKED ISSUE #35294: Various 32-bit failures at MacPython/pandas-wheels
https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=39280&view=logs&j=a846d25a-e32c-5640-1b53-e815fab94407&t=8fc25549-8157-5c9c-f998-0884bd2e8ccb&l=3099

```
FAILED test_venv/lib/site-packages/pandas/tests/window/test_api.py::test_multiple_agg_funcs[rolling-2-expected_vals0]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_apply.py::test_rolling_apply_args_kwargs[args_kwargs0]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_apply.py::test_rolling_apply_args_kwargs[args_kwargs1]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_groupby_rolling[1.0-True]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_groupby_rolling[0.0-False]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_getitem
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_quantile[linear]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_quantile[lower]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_quantile[higher]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_quantile[midpoint]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_quantile[nearest]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_apply[True]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_apply[False]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_grouper.py::TestGrouperGrouping::test_rolling_apply_mutability
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_closed_one_entry_groupby[min]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_closed_one_entry_groupby[max]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_rolling_positional_argument[True-grouping0-_index0]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_rolling_positional_argument[True-grouping1-_index1]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_rolling_positional_argument[False-grouping0-_index0]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_rolling.py::test_rolling_positional_argument[False-grouping1-_index1]
FAILED test_venv/lib/site-packages/pandas/tests/window/test_timeseries_window.py::TestRollingTS::test_groupby_monotonic
FAILED test_venv/lib/site-packages/pandas/tests/window/test_timeseries_window.py::TestRollingTS::test_non_monotonic
```

Just going to skip these for now. Maybe (probably?) some overlap with #35148 @mroeschke. Not sure.

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #35294
- [x] closes #35148
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Tested locally with @TomAugspurger docker container and the tests were working: https://github.com/pandas-dev/pandas/pull/35228#issuecomment-658833239

### CODE REVIEW

File: pandas/core/window/indexers.py [hunk 1]
```diff
@@ -298,7 +299,7 @@ def get_window_bounds(
         window_indicies_start = 0
         for key, indicies in self.groupby_indicies.items():
             if self.index_array is not None:
-                index_array = self.index_array.take(indicies)
+                index_array = self.index_array.take(ensure_platform_int(indicies))
```
Comment (jbrockmendel) [inline pandas/core/window/indexers.py:302]: indicies -> indices?

**Comment by jbrockmendel** [comment]
nice!

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by simonjayhawkins** [comment]
This should backport automatically and fix the failing groupby 32bit tests https://github.com/pandas-dev/pandas/pull/37466#issuecomment-717896083 

after backporting this could also revert #36220

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**